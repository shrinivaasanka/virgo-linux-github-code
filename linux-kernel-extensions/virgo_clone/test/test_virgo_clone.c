#include <stdio.h>
#include <linux/kernel.h>
#include <sys/syscall.h>
#include <unistd.h>

int main(int argc, char* argv[])
{
	/*
	long int ret=syscall(350,"int clone_func(void* args)",NULL,0,NULL);
	*/

	/*
	Mangled name for function prototype "void* virgo_cloud_test(void*)" is sent to virgo_cloudexec kernel service which does
	upcall to userspace to invoke the function (within libvirgo.so) in remote cloud node. Kernel upcall usermode helper 
	requires mangled names for dlsym(). This also is a unique id for a function. Library libvirgo.so needs to be prebuilt
	with all functions that need to be on cloud and should be replicated in all cloud nodes.
	Padded right with 2 whitespaces for ^M removal code in kernel module side.
	- Ka.Shrinivaasan
	*/
	/*long int ret=syscall(350,"_Z16virgo_cloud_testPv  ",NULL,0,NULL);*/
	long int ret=syscall(350,argv[1],NULL,0,NULL);
	return 0;
}
