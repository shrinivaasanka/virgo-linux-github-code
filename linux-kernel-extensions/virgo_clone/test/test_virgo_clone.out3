Jun 28 12:18:34 kashrinivaasan-Inspiron-1545 kernel: [  163.712235] virgo_clone() syscall: created client kernel socket
Jun 28 12:18:34 kashrinivaasan-Inspiron-1545 kernel: [  163.712388] virgocloudexec_recvfrom(): before kernel_recvmsg()
Jun 28 12:18:34 kashrinivaasan-Inspiron-1545 kernel: [  163.712402] virgocloudexec_recvfrom(): kernel_recvmsg() returns len: 500
Jun 28 12:18:34 kashrinivaasan-Inspiron-1545 kernel: [  163.712406] virgocloudexec_recvfrom(): kernel_recvmsg() returns in recv buffer: int clone_func(void* args)
Jun 28 12:18:34 kashrinivaasan-Inspiron-1545 kernel: [  163.712368] virgo_clone() syscall: connected kernel client to virgo cloudexec kernel service
Jun 28 12:18:34 kashrinivaasan-Inspiron-1545 kernel: [  163.712368]  
Jun 28 12:18:34 kashrinivaasan-Inspiron-1545 kernel: [  163.712408] virgo_clone() syscall: sent message: int clone_func(void* args) 
Jun 28 12:18:34 kashrinivaasan-Inspiron-1545 kernel: [  163.712413] print_buffer(): 
Jun 28 12:18:34 kashrinivaasan-Inspiron-1545 kernel: [  163.712416] in
Jun 28 12:18:34 kashrinivaasan-Inspiron-1545 kernel: [  163.712420] t 
Jun 28 12:18:34 kashrinivaasan-Inspiron-1545 kernel: [  163.712424] 
Jun 28 12:18:34 kashrinivaasan-Inspiron-1545 kernel: [  163.712427] virgocloudexec_recvfrom(): kernel_recvmsg() le32 to cpu int clone_func(void* args)
Jun 28 12:18:34 kashrinivaasan-Inspiron-1545 kernel: [  163.712430] virgocloudexec_recvfrom(): cloneFunction : int clone_func(void* args) 
Jun 28 12:18:34 kashrinivaasan-Inspiron-1545 kernel: [  163.712489] virgocloudexec_recvfrom(): clone thread woken_up : 1
Jun 28 12:18:34 kashrinivaasan-Inspiron-1545 kernel: [  163.712493] clone_func(): executing the virgo_clone() syscall function parameter in cloud 
Jun 28 12:18:34 kashrinivaasan-Inspiron-1545 kernel: [  163.712497] virgocloudexec_sendto(): before kernel_sendmsg() for send buffer: virgo_cloudexec_sendto(): cloudclonethread executed for clone_func(), sending message to virgo_clone() remote syscall client
Jun 28 12:18:34 kashrinivaasan-Inspiron-1545 kernel: [  163.712497] 
Jun 28 12:18:34 kashrinivaasan-Inspiron-1545 kernel: [  163.712529] virgocloudexec_sendto(): kernel_sendmsg() returns ret: 500
Jun 28 12:18:34 kashrinivaasan-Inspiron-1545 kernel: [  163.712533] virgocloudexec_sendto(): Shut down Kernel Side Client Socket with SOCK_WAKE_URG after sendmsg 
Jun 28 12:18:34 kashrinivaasan-Inspiron-1545 kernel: [  163.712559] virgocloudexec_sendto(): sock_release invoked on client socket 
Jun 28 12:18:34 kashrinivaasan-Inspiron-1545 kernel: [  163.712600] virgo_clone() syscall: received message: virgo_cloudexec_sendto(): cloudclonethread executed for clone_func(), sending message to virgo_clone() remote syscall client
Jun 28 12:18:34 kashrinivaasan-Inspiron-1545 kernel: [  163.712600]  
Jun 28 12:18:34 kashrinivaasan-Inspiron-1545 kernel: [  163.712606] virgo_clone() syscall: le32_to_cpus(buf): virgo_cloudexec_sendto(): cloudclonethread executed for clone_func(), sending message to virgo_clone() remote syscall client
Jun 28 12:18:34 kashrinivaasan-Inspiron-1545 kernel: [  163.712606]  
Jun 28 12:18:34 kashrinivaasan-Inspiron-1545 kernel: [  163.712614] virgo_clone() syscall: virgo_clone() client socket_release() invoked
Jun 28 12:19:04 kashrinivaasan-Inspiron-1545 kernel: [  193.899339] virgocloudexec_recvfrom(): before kernel_recvmsg()
Jun 28 12:19:17 kashrinivaasan-Inspiron-1545 kernel: [  206.874082] virgocloudexec_recvfrom(): kernel_recvmsg() returns len: 27
Jun 28 12:19:17 kashrinivaasan-Inspiron-1545 kernel: [  206.874090] virgocloudexec_recvfrom(): kernel_recvmsg() returns in recv buffer: int function1(void* args)
Jun 28 12:19:17 kashrinivaasan-Inspiron-1545 kernel: [  206.874090] 
Jun 28 12:19:17 kashrinivaasan-Inspiron-1545 kernel: [  206.874095] print_buffer(): 
Jun 28 12:19:17 kashrinivaasan-Inspiron-1545 kernel: [  206.874097] in
Jun 28 12:19:17 kashrinivaasan-Inspiron-1545 kernel: [  206.874102] t 
Jun 28 12:19:17 kashrinivaasan-Inspiron-1545 kernel: [  206.874106] 
Jun 28 12:19:17 kashrinivaasan-Inspiron-1545 kernel: [  206.874109] virgocloudexec_recvfrom(): kernel_recvmsg() le32 to cpu int function1(void* args)
Jun 28 12:19:17 kashrinivaasan-Inspiron-1545 kernel: [  206.874109] 
Jun 28 12:19:17 kashrinivaasan-Inspiron-1545 kernel: [  206.874113] virgocloudexec_recvfrom(): cloneFunction : int function1(void* args)
Jun 28 12:19:17 kashrinivaasan-Inspiron-1545 kernel: [  206.874113]  
Jun 28 12:19:17 kashrinivaasan-Inspiron-1545 kernel: [  206.874253] virgocloudexec_recvfrom(): clone thread woken_up : 1
Jun 28 12:19:17 kashrinivaasan-Inspiron-1545 kernel: [  206.874258] virgocloudexec_sendto(): before kernel_sendmsg() for send buffer: virgo_cloudexec_sendto(): cloudclonethread executed for clone_func(), sending message to virgo_clone() remote syscall client
Jun 28 12:19:17 kashrinivaasan-Inspiron-1545 kernel: [  206.874258] 
Jun 28 12:19:17 kashrinivaasan-Inspiron-1545 kernel: [  206.874299] virgocloudexec_sendto(): kernel_sendmsg() returns ret: 500
Jun 28 12:19:17 kashrinivaasan-Inspiron-1545 kernel: [  206.874303] virgocloudexec_sendto(): Shut down Kernel Side Client Socket with SOCK_WAKE_URG after sendmsg 
Jun 28 12:19:17 kashrinivaasan-Inspiron-1545 kernel: [  206.874328] virgocloudexec_sendto(): sock_release invoked on client socket 
Jun 28 12:19:17 kashrinivaasan-Inspiron-1545 kernel: [  206.874779] clone_func(): executing the virgo_clone() syscall function parameter in cloud 

