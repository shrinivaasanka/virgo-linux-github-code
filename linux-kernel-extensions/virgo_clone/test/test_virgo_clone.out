tcp        0      0 127.0.0.1:34404         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34399         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34412         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:60992         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34400         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34402         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34430         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:10000         127.0.0.1:34248         ESTABLISHED -               
tcp        0      0 127.0.0.1:34427         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34426         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34394         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34431         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34410         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:60991         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34423         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34401         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:10000         127.0.0.1:60989         ESTABLISHED -               
tcp        0      0 127.0.0.1:34403         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:10000         127.0.0.1:60991         ESTABLISHED -               
tcp        0      0 127.0.0.1:34415         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34419         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34406         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34429         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34248         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34409         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34413         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34428         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34420         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34411         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34408         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34417         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34418         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34261         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34414         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:10000         127.0.0.1:60992         ESTABLISHED -               
tcp        0      0 127.0.0.1:60989         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34405         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34416         127.0.0.1:10000         ESTABLISHED -               
tcp        0      0 127.0.0.1:34407         127.0.0.1:10000         ESTABLISHED -               
